lib/shader-cache.js

```diff
@@ -47,14 +48,41 @@ function ContextCache(gl) {
 
 var proto = ContextCache.prototype
 
+function formatCompileError(errLog, src) {
+    var errorStrings = errLog.split('\n');
+    var errors = {};
+    for (var i = 0; i < errorStrings.length; i++) {
+        var errorString = errorStrings[i];
+        var lineNo = parseInt(errorString.split(':')[2]);
+        errors[lineNo] = errorString;
+    }
+    var lines = src.split('\n');
+    var fmt = "";
+    for (var i = 0; i < lines.length; i++) {
+        var line = lines[i];
+        fmt += sprintf("%4d: %s\n", i+1, line);
+        if (errors[i+1]) {
+            var e = errors[i+1];
+            e = e.substr(e.split(':', 3).join(':').length + 1).trim();
+            fmt += sprintf('^^^^: %s\n', e);
+        }
+    }
+    return fmt;
+}
+
```

Comment:
This might be useful as an independent package — that way it's easy for others to reuse in other projects too :)